strawberryfields/backends/bosonicbackend/backend.py

```diff
@@ -140,9 +142,9 @@ def run_prog(self, prog, **kwargs):
                 if val is not None:
                     for i, r in enumerate(cmd.reg):
                         if r.ind not in self.ancillae_samples_dict.keys():
-                            self.ancillae_samples_dict[r.ind] = [val[:, i]]
+                            self.ancillae_samples_dict[r.ind] = [val]
```

Comment:
How come this was changed? Guess it's connected to adding `.T` and adjusting dims down below.

Reply:
There were several iterations of this, but this form is so the dimensions of the output match with what is expected by the Results class. The number of rows in the samples needs to match the number of shots, and the number of columns has to match the number of modes.